flink-ml-lib/src/main/java/org/apache/flink/ml/feature/bucketizer/Bucketizer.java

```diff
@@ -0,0 +1,179 @@
+                new RowTypeInfo(
+                        ArrayUtils.addAll(inputTypeInfo.getFieldTypes(), outputTypes),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), getOutputCols()));
+
+        DataStream<Row> result =
+                tEnv.toDataStream(inputs[0])
+                        .flatMap(
+                                new FindBucketFunction(inputCols, splitsArray, getHandleInvalid()),
+                                outputTypeInfo);
+        return new Table[] {tEnv.fromDataStream(result)};
+    }
+
+    /** Finds the bucket index for each continuous feature of an input data point. */
+    private static class FindBucketFunction implements FlatMapFunction<Row, Row> {
+        private final String[] inputCols;
+        private final String handleInvalid;
+        private final Double[][] splitsArray;
+
+        public FindBucketFunction(
+                String[] inputCols, Double[][] splitsArray, String handleInvalid) {
+            this.inputCols = inputCols;
+            this.splitsArray = splitsArray;
+            this.handleInvalid = handleInvalid;
+        }
+
+        @Override
+        public void flatMap(Row value, Collector<Row> out) {
+            Row outputRow = new Row(inputCols.length);
+
+            for (int i = 0; i < inputCols.length; i++) {
+                double feature = ((Number) value.getField(inputCols[i])).doubleValue();
+                Double[] splits = splitsArray[i];
+                boolean isInvalid = false;
+
+                if (!Double.isNaN(feature)) {
+                    int index = Arrays.binarySearch(splits, feature);
+                    if (index >= 0) {
+                        if (index == inputCols.length - 1) {
+                            index--;
+                        }
+                        outputRow.setField(i, index);
+                    } else {
+                        index = -index - 1;
+                        if (index == 0 || index == inputCols.length) {
+                            isInvalid = true;
+                        } else {
+                            outputRow.setField(i, index - 1);
+                        }
+                    }
+                } else {
+                    isInvalid = true;
+                }
+
+                if (isInvalid) {
+                    switch (handleInvalid) {
+                        case BucketizerParams.ERROR_INVALID:
+                            throw new RuntimeException(
+                                    "The input contains invalid value. See "
+                                            + BucketizerParams.HANDLE_INVALID
+                                            + " parameter for more options.");
+                        case BucketizerParams.SKIP_INVALID:
+                            return;
+                        case BucketizerParams.KEEP_INVALID:
+                            outputRow.setField(i, splits.length - 1);
+                            break;
```

Comment:
This should be `continue`.

Reply:
Hmm, could you explain a bit more about this?

Reply:
For example, If the input have three columns, `[a, b, c]`. `a` and `c` are valid values mapping to `a*` and `c*`, while `b` is an invalid value. In this case the operator should output `[a*, splits.length-1, c*]`, but if the code here is `break`,  the operator would actually output `[a*,  splits.length-1, null]`.

Reply:
@yunfengzhou-hub by the `break` here, the code breaks out of the switch loop, and the for loop continues. So in the case described above, the current implementation would still output `[a*, splits.length-1, c*]`.